artifact/compressor_lzma.go

```diff
@@ -11,15 +11,16 @@
 //    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 //    See the License for the specific language governing permissions and
 //    limitations under the License.
-//go:build !nolzma && cgo
-// +build !nolzma,cgo
+//go:build !nolzma && !cgo
+// +build !nolzma,!cgo
```

Comment:
I am actually in favor of replacing LZMA compression from shared to static linked version (this one, and drop the old code) instead of having the two with compile flags.

Pros:
* Safer testing (as it stands now, for example, the acceptance and integration tests will only test the cgo version)
* One less OS dependency (`xz` is a pretty safe dependency, but still)

Cons:
* Bigger binary size (but it can still be opted out with `nolzma` flag)

Reply:
Agree, let's just ditch the old one. Size is even less important now because the C++ client does not use mender-artifact to parse. Then we can remove `cgo` completely from the equation here.

Reply:
Done